File: converter/utilities/generate_token.js
const jwt = require('jsonwebtoken');
const fs = require('fs');
const path = require('path');

// Apple Developer account details
const TEAM_ID = '732N38L7AM';
const KEY_ID = '943ZMY4RS6';
const PRIVATE_KEY_PATH = path.join(__dirname, 'apple_auth_key.p8');

try {
  // Read the private key
  const privateKey = fs.readFileSync(PRIVATE_KEY_PATH, 'utf8');

  // Generate token with correct claims
  const token = jwt.sign(
    {
      iss: TEAM_ID, // Your Apple Developer Team ID
      iat: Math.floor(Date.now() / 1000), // Current time in seconds
      exp: Math.floor(Date.now() / 1000) + (60 * 60 * 24 * 180), // 180 days from now
      aud: 'music' // Must be 'music' for MusicKit
    },
    privateKey,
    {
      algorithm: 'ES256', // Required algorithm
      header: {
        alg: 'ES256',
        kid: KEY_ID // Your MusicKit Key ID
      }
    }
  );

  console.log('\nGenerated Developer Token:');
  console.log('=========================');
  console.log(token);
  console.log('\nToken Details:');
  console.log('=============');

  // Decode and display token info
  const decoded = jwt.decode(token, { complete: true });
  console.log('Header:', JSON.stringify(decoded.header, null, 2));
  console.log('Payload:', JSON.stringify(decoded.payload, null, 2));
  console.log('\nExpiration:', new Date(decoded.payload.exp * 1000).toLocaleString());
} catch (error) {
  console.error('Error generating token:', error.message);
  if (error.code === 'ENOENT') {
    console.error(`\nPrivate key file not found at: ${PRIVATE_KEY_PATH}`);
    console.error('Make sure your .p8 file is in the correct location.');
  }
  process.exit(1);
}

File: converter/utilities/generate_token.py
import jwt
import time
from pathlib import Path

def generate_token():
    # Your credentials
    key_id = 'WTM4GAXM4P'
    team_id = '732N38L7AM'
    private_key_path = Path(__file__).parent / 'apple_auth_key.p8'
    
    # Read the private key
    with open(private_key_path, 'r') as key_file:
        private_key = key_file.read()
    
    # Generate the token
    headers = {
        'alg': 'ES256',
        'kid': key_id
    }
    
    payload = {
        'iss': team_id,
        'iat': int(time.time()),
        'exp': int(time.time() + 15777000)  # 6 months from now
    }
    
    # Generate the token
    token = jwt.encode(
        payload,
        private_key,
        algorithm='ES256',
        headers=headers
    )
    
    print("Generated Developer Token:")
    print(token)
    print("\nThis token is valid for 6 months.")
    return token

if __name__ == '__main__':
    generate_token() 
